node/sc/bridge_manager.go

```diff
@@ -311,7 +314,19 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
 		}
 		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
 	case ERC721:
-		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.Uri, ev.ExtraData)
+		// get URI of the ERC721
+		var uri string
+		erc721, err := scnft.NewERC721Metadata(ev.TokenAddress, bi.counterpartBackend)
+		if err != nil {
+			logger.Warn("Failed to get erc721 token instance", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain)
+		} else {
+			uri, err = erc721.TokenURI(nil, ev.ValueOrTokenId)
+			if err != nil {
+				logger.Warn("Failed to get URI of the ERC721 token", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain, "tokenId", ev.ValueOrTokenId.String())
+			}
+		}
+
+		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, uri, ev.ExtraData)
```

Comment:
Does this matter if the err is not nil?

Reply:
Yes. If this handling fails, bridge node should retry to handle it again. So this code return the error.

Reply:
Sorry. The comment is wrong. "323: uri, err = erc721.TokenURI (nil, ev.ValueOrTokenId)" was asking if you can use uri when err is not nil.

Reply:
I thought if the erc721 doen't support URI (metadata), skipping is ok.
But this code can skip other error cases like communication error. So I will try to check specific `revert` error. :) Thanks, pal.

Reply:
I rebased it. :)